internal/locate/region_cache.go

```diff
@@ -726,9 +726,46 @@ func NewRegionCache(pdClient pd.Client, opt ...RegionCacheOpt) *RegionCache {
 		// cache GC is incompatible with cache refresh
 		c.bg.schedule(c.gcRoundFunc(cleanRegionNumPerRound), cleanCacheInterval)
 	}
+	c.bg.schedule(
+		func(ctx context.Context, _ time.Time) bool {
+			refreshFullStoreList(ctx, c.PDClient(), c.stores)
+			return false
+		}, refreshStoreListInterval,
+	)
 	return c
 }
 
+// Try to refresh full store list. Errors are ignored.
+func refreshFullStoreList(ctx context.Context, pdClient pd.Client, stores storeCache) {
+	storeList, err := pdClient.GetAllStores(ctx)
+	if err != nil {
+		logutil.Logger(ctx).Info("refresh full store list failed", zap.Error(err))
+		return
+	}
+	for _, store := range storeList {
+		_, exist := stores.get(store.GetId())
+		if exist {
+			continue
+		}
+		s := stores.getOrInsertDefault(store.GetId())
+		if store == nil || store.GetState() == metapb.StoreState_Tombstone {
+			s.setResolveState(tombstone)
+			continue
+		}
+		addr := store.GetAddress()
+		if addr == "" {
+			continue
```

Comment:
Better to explain in the comments when the `pdClient.GetAllStores(ctx)` may return a store list containg `nil` store and `emptry address` and how should they be treated. For example, dose it mean the returned  `store list` is  invalid and `pdClient.GetAllStores(ctx)` should be retried when there invalid element in the list?

Reply:
I believe that `GetAllStores` is supposed to return only stores in Up and Offline states. This additional check serves as a defensive measure and ensures consistency with the store resolve procedure